### PR TITLE
Add the Two Jug Water Puzzle

### PR DESCRIPTION
**Pull request by vivek09thakur**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: TwoJugWaterPuzzle/00_two_jug_water_puzzle.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+"Implementation of the two jug water puzzle."
+
+
+def gcd(a, b):
```
Comment (cclauss) [inline TwoJugWaterPuzzle/00_two_jug_water_puzzle.py:4]: Let's help the reader to understand...
```suggestion
def greatest_common_denominator(a, b):
```

File: TwoJugWaterPuzzle/00_two_jug_water_puzzle.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    "Return the greatest common divisor of a and b."
+    if b == 0:
+        return a
+    else:
+        return gcd(b, a % b)
```
Comment (cclauss) [inline TwoJugWaterPuzzle/00_two_jug_water_puzzle.py:9]:
```suggestion
    if b == 0:
    return a if b == 0 else gcd(b, a % b)
```

File: TwoJugWaterPuzzle/00_two_jug_water_puzzle.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+def pour(toJugCap, fromJugcap, d):
+    """
+    fromCap -- Capacity of jug from which
+              water is poured
+    toCap   -- Capacity of jug to which
+              water is poured
+     d       -- Amount to be measured
+    """
+    fromJug = fromJugcap
+    toJug = 0
```
Comment (cclauss) [inline TwoJugWaterPuzzle/00_two_jug_water_puzzle.py:21]: This is Python, not JavaScript.
```suggestion
    from_jug = from_jug_capacity
    to_jug = 0
```

File: TwoJugWaterPuzzle/00_two_jug_water_puzzle.py (hunk 1 above)
Comment (vivek09thakur) [reply TwoJugWaterPuzzle/00_two_jug_water_puzzle.py]: okay I will make sure this

File: two_jug_water_puzzle/two_jug_water_puzzle.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    step = 1
+    while fromJug != d and toJug != d:
+        temp = min(fromJug, toJugCap - toJug)
+        toJug = toJug + temp
+        fromJug = fromJug - temp
+        step = step + 1
+        if fromJug == d or toJug == d:
+            break
+        if fromJug == 0:
+            fromJug = fromJugCap
```
[the remark this answers is not in the document]
Comment (vivek09thakur) [reply two_jug_water_puzzle/two_jug_water_puzzle.py]: i resolved the naming issues

**Pull request closed without merging**